### PR TITLE
Adds S3Persistance plugin

### PR DESCRIPTION
**Pull request by omenocal**
This enables developers to storage the user's session attributes in a file in an S3 bucket instead of DynamoDB. It's specially helpful if you are building your skill with the Alexa-Hosted feature. More info here:

https://developer.amazon.com/docs/hosted-skills/build-a-skill-end-to-end-using-an-alexa-hosted-skill.html#persistence

### CODE REVIEW

File: docs/plugins.rst [hunk 1]
```diff
@@ -155,3 +155,37 @@ Usage
...
 
     plugins.autoLoad(app, { adapter });
+
+
+
+
+S3Persistence plugin
+--------------------
+
+It stores the user's session attributes in a file in an S3 bucket.
```
Comment (rmberrios) [inline docs/plugins.rst]: add a comment that instructs developers to use this functionality with Alexa hosting service.

Comment (wuelcas) [reply docs/plugins.rst]: I agree, you can add some recommendations on which situation this plugin is most useful.

Comment (omenocal) [reply docs/plugins.rst]: @rmberrios @wuelcas done!

File: src/plugins/s3-persistence.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import { S3 } from "aws-sdk";
+import * as _ from "lodash";
+import * as path from "path";
+import { VoxaApp } from "../VoxaApp";
+import { IVoxaEvent } from "../VoxaEvent";
+import { IVoxaReply } from "../VoxaReply";
+
+let defaultConfig: any = {};
+
+export function s3Persistence(voxaApp: VoxaApp, config: any) {
```
Comment (armonge) [inline src/plugins/s3-persistence.ts]: Would be great to have an actual interface for the config

File: src/plugins/s3-persistence.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    const body = result.Body ? result.Body.toString() : "{}";
+    const data = JSON.parse(body);
+
+    voxaEvent.log.debug("Data fetched:", { data });
+    voxaEvent.model.user = data;
+    return voxaEvent;
+  } catch (error) {
+    voxaEvent.log.error(error);
```
Comment (armonge) [inline src/plugins/s3-persistence.ts]: voxa itself has automatic logging of uncaught exceptions, do we really need to log it here too?

File: src/plugins/s3-persistence.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+      Bucket: defaultConfig.bucketName,
+      Key: objectId,
+    };
+    let result: S3.GetObjectOutput = {};
+
+    try {
+      result = await defaultConfig.s3Client.getObject(getParams).promise();
+    } catch (error) {
+      if (error.code !== "NoSuchKey") {
+        voxaEvent.log.error(error);
```
Comment (armonge) [inline src/plugins/s3-persistence.ts]: actually logging things 3 times :thinking:

File: test/dialogflow/google/DigitalGoods.spec.ts [hunk 5]
```diff
@@ -136,7 +143,9 @@ describe("DigitalGoods", () => {
       };
     });
 
-    const reply = await googleAssistantPlatform.execute(buyProductIntent);
+    const googleAssistantPlatformWithKeyFile = new GoogleAssistantPlatform(voxaApp, configKeyFile);
```
Comment (armonge) [inline test/dialogflow/google/DigitalGoods.spec.ts]: this seems like something that should go on another PR

File: test/plugins/s3-persistence.spec.ts [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+import { expect } from "chai";
+import * as simple from "simple-mock";
+import { AlexaPlatform, plugins, VoxaApp, VoxaEvent } from "../../src";
+import { AlexaRequestBuilder } from "../tools";
+import { variables } from "../variables";
+import { views } from "../views";
+
+const rb = new AlexaRequestBuilder("user-xyz");
+
+describe("S3Persistence plugin", () => {
```
Comment (armonge) [inline test/plugins/s3-persistence.spec.ts]: Would be great if we did the testing using something like https://github.com/jamhall/s3rver

File: src/plugins/s3-persistence.ts [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { S3 } from "aws-sdk";
```
Comment (armonge) [inline src/plugins/s3-persistence.ts]: I don't see aws-sdk as dependency in package.json
